src/Properties/ModelCastHelper.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace NunoMaduro\Larastan\Properties;
+
+use Illuminate\Contracts\Database\Eloquent\Castable;
+use Illuminate\Contracts\Database\Eloquent\CastsAttributes;
+use Illuminate\Contracts\Database\Eloquent\CastsInboundAttributes;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use PHPStan\Reflection\ParameterReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Type\Accessory\AccessoryNumericStringType;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\BooleanType;
+use PHPStan\Type\Constant\ConstantIntegerType;
+use PHPStan\Type\FloatType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeCombinator;
+
+class ModelCastHelper
+{
+    public function __construct(protected ReflectionProvider $reflectionProvider)
+    {
+    }
+
+    public function getReadableType(string $cast, Type $originalType): Type
+    {
+        $cast = Str::before($cast, ':');
+
+        $attributeType = match ($cast) {
+            'int', 'integer' => new IntegerType(),
+            'real', 'float', 'double' => new FloatType(),
+            'decimal' => new AccessoryNumericStringType(),
```

Comment:
Accessory types cannot be used standalone. You need to do `TypeCombinator::intersect(new StringType(), new AccessoryNumericStringType())` here